windows/src/developer/TIKE/compile/MergeKeyboardInfo.pas

```diff
@@ -288,22 +290,39 @@ function TMergeKeyboardInfo.LoadKMPFile: Boolean;
         end;
       end;
 
-      FKMXTempFile := TTempFileManager.Get('.kmx');
-      try
-        for i := 0 to High(Zip.FileNames) do
-        begin
-          if SameText(ExtractFileExt(Zip.FileNames[i]), '.kmx') then
-          begin
-            SetLength(FKMXFiles, Length(FKMXFiles)+1);
-
-            SaveMemberToFile(Zip.FileNames[i], FKMXTempFile.Name);
 
-            FKMXFiles[High(FKMXFiles)].Filename := Zip.FileNames[i];
-            GetKeyboardInfo(FKMXTempFile.Name, False, FKMXFiles[High(FKMXFiles)].Info, False);
+      for i := 0 to FKMPInfFile.Keyboards.Count - 1 do
+      begin
+        FKMXTempFile := TTempFileManager.Get('.kmx');
+        FJSTempFile := TTempFileManager.Get('.js');
+        try
+          for j := 0 to High(Zip.FileNames) do
+          begin
+            // Add the KMX to FPackageKMXFileinfos
+            if SameText(Zip.FileName[j], FKMPInfFile.Keyboards[i].ID + '.kmx') then
```

Comment:
This line breaks legacy keyboards where the .kmx file doesn't match the keyboard ID name.